forms/src/main/java/org/devgateway/toolkit/forms/wicket/page/lists/AbstractListPage.java

```diff
@@ -224,6 +207,40 @@ protected void onSubmit(final AjaxRequestTarget target) {
         add(topEditPageLink);
     }
 
+    private void initializeColumns() {
+        columns.add(new AbstractColumn<T, String>(new Model<>("#")) {
+            @Override
+            public void populateItem(final Item<ICellPopulator<T>> cellItem,
+                    final String componentId,
+                    final IModel<T> model) {
+                final OddEvenItem<?> oddEvenItem = (OddEvenItem<?>) cellItem.getParent().getParent();
+                final long index = WebConstants.PAGE_SIZE * dataTable.getCurrentPage() + oddEvenItem.getIndex() + 1;
+                cellItem.add(new Label(componentId, index));
+            }
+        });
+
+        addColumns();
+
+        // add the 'Edit' button
+        if (hasEditPage) {
+            columns.add(new AbstractColumn<T, String>(new StringResourceModel("actionsColumn", this, null)) {
+                private static final long serialVersionUID = -7447601118569862123L;
+
+                @Override
+                public void populateItem(final Item<ICellPopulator<T>> cellItem, final String componentId,
+                        final IModel<T> model) {
+                    cellItem.add(getActionPanel(componentId, model));
+                }
+            });
+        }
+    }
+
+    protected abstract void addColumns();
+
+    public AjaxFallbackBootstrapDataTable<T, String> getDataTable() {
```

Comment:
use `protected` getters when replacing access to protected fields. also If getter is not to be changed, use `protected final`

Reply:
Done.